pagopa/io-pay
# [#175848172] Integrated PM Client

**Description**
This PR proposes to integrate in `io-pay` the PM client developed for IO APP.

**Proposed Changes**
- Added the PM client and its dependencies (based on old PM spec)
- Improved test suite for `fetch utils`
- Limited but extensible test suite for PM Client
- Removed in `defaultRetryingFetch` the dependency from a custom `whatwg-fetch` module

**How to test**
I added unit tests for `fetch utils` and `PM client`. Please note that to avoid mocking fetches I used the following approach https://kentcdodds.com/blog/stop-mocking-fetch , based on an `express` app which better suits our stack as an alternative to the `msw` package
